receiver/hostmetricsreceiver/config.go

```diff
@@ -15,20 +15,72 @@
 package hostmetricsreceiver
 
 import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cast"
+
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/receiver/hostmetricsreceiver/internal"
 	"go.opentelemetry.io/collector/receiver/scraperhelper"
 )
 
+const (
+	scrapersKey = "scrapers"
+)
+
 // Config defines configuration for HostMetrics receiver.
 type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
 	Scrapers                                map[string]internal.Config `mapstructure:"-"`
 }
 
 var _ config.Receiver = (*Config)(nil)
+var _ config.Unmarshable = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid
 func (cfg *Config) Validate() error {
 	return nil
 }
+
+// Unmarshal a config.Parser into the config struct.
+func (cfg *Config) Unmarshal(componentParser *config.Parser) error {
+	// load the non-dynamic config normally
+	err := componentParser.Unmarshal(cfg)
+	if err != nil {
+		return err
```

Comment:
This (and other error modes on the `customUnmarshaler`s) was untested before, I can add tests for this but I think it can be done on a different PR.